### PR TITLE
[SYCL] Add more GPU options to available features

### PR DESCRIPTION
**Pull request by myler**
add GPU FAMILY to available features and specify ESIMD raw send tests to gpu-intel-gen9

### CODE REVIEW

**Comment by myler** [comment]
Add @pvchupin @steffenlarsen @kbobrovs for code review.

**Comment by kbobrovs** [comment]
ESIMD LGTM.
One note:
I can see that Jenkins CI was done on a platform not supporting FP64 (TGLLP?), as many tests are skipped as UNSUPPORTED. I hope we run regular CI on GPU platforms with FP64 supported, otherwise lots of important GPU tests are not run.
@pvchupin, JFYI.

**Review by kbobrovs — DISMISSED** [review]
ESIMD LGTM

**Comment by myler** [comment]
> ESIMD LGTM. One note: I can see that Jenkins CI was done on a platform not supporting FP64 (TGLLP?), as many tests are skipped as UNSUPPORTED. I hope we run regular CI on GPU platforms with FP64 supported, otherwise lots of important GPU tests are not run. @pvchupin, JFYI.

I think not only `aspect-fp64` but other identifiers, such as `gpu-intel-*` are not able to be handled by Jenkins CI, because Jenkins CI  doesn't  know what `aspect-fp64` is. Then many tests returned UNSUPPORTED.

If this change got approval, or more tests want to run via Jenkins CI, it also needs to update, @DoyleLi could you help on this?

- If device is one of CPU, ACC, or HOST, then `double` type is native supported, so by default, `python lit.py` needs an additional option `-Daspect-fp64=1`
- If device is GPU, then check whether does GPU HW support `double` type, if yes, `python lit.py` needs an additional option `-Daspect-fp64=1`, e.g. on Gen9 and older GPUs.

In internal infrastructure, we check the dispatched HW features and provide additional options to `python lit.py`

**Comment by kbobrovs** [comment]
> I think not only aspect-fp64 but other identifiers, such as gpu-intel-* are not able to be handled by Jenkins CI, because Jenkins CI doesn't know what aspect-fp64 is. Then many tests returned UNSUPPORTED.

Right. My point is that absence of fp64 disables particularly big number of important tests, so it is highly desirable to run CI on GPU with fp64 support.

**Comment by pvchupin** [comment]
@myler, please fix PR title and description.

File: SYCL/Basic/buffer/buffer.cpp
Comment (pvchupin) [inline SYCL/Basic/buffer/buffer.cpp]: It doesn't look correct to me for this test and many others.
We probably have only a small portion of test that requires aspect-fp64. In such case we should split the test and have buffer.cpp without this requirement and buffer-aspect-fp64 for the part where it's required.

@intel/llvm-reviewers-runtime should review carefully.

**Review by pvchupin — CHANGES_REQUESTED** [review]

File: SYCL/Basic/buffer/buffer.cpp
Comment (myler) [reply SYCL/Basic/buffer/buffer.cpp]: To split `buffer.cpp` to `buffer.cpp` and `buffer-aspect-fp64.cpp` with `// REQUIRES: aspect-fp64` is more reasonable.  But to split all of them (up to 50 test files), I need some time and effort to do this.

Or we may update test source with `device.has(sycl::aspect::fp64)` or other feature checkers to strip type coverage from test logic?

Comment (pvchupin) [reply SYCL/Basic/buffer/buffer.cpp]: I'm not sure what's the current state of device.has() support. I'm not sure if we have everything needed to mark code portion within a single source file. That would make sense if supported.

Comment (myler) [reply SYCL/Basic/buffer/buffer.cpp]: If no objection, I will try to split 'double' type code to new `.cpp` file with `-aspect-fp64` suffix, and will create separate PRs for different code owner review.

Comment (myler) [reply SYCL/Basic/buffer/buffer.cpp]: Revert `aspect-fp64` related changes, only keep Intel GPU FAMILY availability for restricting ESIMD raw_send tests.

**Comment by pvchupin** [comment]
@myler, please update PR description and fix title.

File: SYCL/ESIMD/histogram_raw_send.cpp
Comment (pvchupin) [inline SYCL/ESIMD/histogram_raw_send.cpp]: is "gpu" required to stay on the list here?
Should it be gpu-intel-gen9 only?

Comment (v-klochkov) [reply SYCL/ESIMD/histogram_raw_send.cpp]: IMO, gpu-intel-gen9 should be enough.

**Comment by myler** [comment]
The failure is not related to this PR, it caused by #888 
```
[2022-06-14T00:41:30.358Z] ********************
[2022-06-14T00:41:30.358Z] Failed Tests (1):
[2022-06-14T00:41:30.358Z]   SYCL :: Basic/aspects.cpp
```

**Comment by myler** [comment]
@pvchupin may I ask you to review again?

**Review by pvchupin — APPROVED** [review]

**Review by v-klochkov — APPROVED** [review]

**Comment by myler** [comment]
Hi @v-klochkov,

Seems I do not have permission to merge to this protected branch. May I ask you to help to merge this PR or is it possible to let me know how to apply the permission?

**Pull request merged**
